pkg/server/server.go

```diff
@@ -129,6 +132,20 @@ func NewServer(ctx context.Context, cfg *config.Config, logger *zap.Logger, name
 			return
 		}
 
+		if errors.Is(dirErr, os.ErrNotExist) {
```

Comment:
Maybe it's a better idea to check whether there are any namespaces after etcd starts? Is it easy to implement?
`work` exists doesn't really mean the etcd database exists.

Reply:
Yes, we can implement it with an extra RPC.

Reply:
Why change it back to check the dir?

Reply:
Because you said, one may don't want the default fallback. IMO, that means the proxy should fail if no namespace matched. But the default fallback works by detecting if there is any namespace. So there is always a default fallback.

I've changed it back to ensure it is only ran once. But after a second thought, I think we can solve this problem by support PD per namespace later. An empty namespace will prevent client connections, and only a non-empty/effective fallback will serve as the real "fallback".